tests/queries/0_stateless/02710_allow_suspicious_indices.sql

```diff
@@ -0,0 +1,13 @@
+DROP TABLE IF EXISTS allow_suspicious_indices;
+CREATE TABLE allow_suspicious_indices (id UInt32) ENGINE = MergeTree() order by (id, id) settings allow_suspicious_indices = 1;
+DROP TABLE IF EXISTS allow_suspicious_indices;
+CREATE TABLE allow_suspicious_indices (id UInt32) ENGINE = MergeTree() order by (id, id) settings allow_suspicious_indices = 0;  -- { serverError BAD_ARGUMENTS }
+DROP TABLE IF EXISTS allow_suspicious_indices;
+CREATE TABLE allow_suspicious_indices (id UInt32, index idx (id, id) TYPE minmax GRANULARITY 1) ENGINE = MergeTree() order by id settings allow_suspicious_indices = 1;
+DROP TABLE IF EXISTS allow_suspicious_indices;
+CREATE TABLE allow_suspicious_indices (id UInt32, index idx (id, id) TYPE minmax GRANULARITY 1) ENGINE = MergeTree() order by id settings allow_suspicious_indices = 0;  -- { serverError BAD_ARGUMENTS }
+DROP TABLE IF EXISTS allow_suspicious_indices;
+CREATE TABLE allow_suspicious_indices (id UInt32) ENGINE = MergeTree() order by id + 1 settings allow_suspicious_indices = 0;
+ALTER TABLE allow_suspicious_indices ADD COLUMN `id2` UInt32, MODIFY ORDER BY (id + 1, id2 + 1, id2 + 1);  -- { serverError BAD_ARGUMENTS }
+DROP TABLE IF EXISTS allow_suspicious_indices;
```

Comment:
Wondering if keys `id + 1` & `1 + id` would also be blocked? But I agree that determining the equivalence of  two expressions will be difficult.

Reply:
Table will be created without exception.